cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -225,24 +263,63 @@ async def prompt_async(
-    ) -> float:
-        """Gets confidence score for prompt-response pair.
+        self,
+        prompt: Union[str, Collection[str]],
+        response: Union[str, Collection[str]],
+        options: Union[None, TLMOptions, Collection[Union[TLMOptions, None]]] = None,
+        timeout: Optional[float] = None,
+        retries: int = 1,
+    ) -> Union[float, List[float]]:
+        """Gets confidence score for prompt-response pair(s).
 
         Args:
-            prompt: prompt for the TLM
-            response: response for the TLM to evaluate
-            options (Optional[TLMOptions]): options to parameterize TLM with. Defaults to None.
+            prompt (str | Collection[str]): prompt (or list/iterable of multiple prompts) for the TLM
+            response (str | Collection[str]): response (or list/iterable of multiple responses) for the TLM to evaluate
+            options (None | TLMOptions | List[TLMOptions  |  None], optional): list of options (or instance of options) to pass to get confidence score method. Defaults to None.
+            timeout (Optional[float], optional): maximum allowed time (in seconds) to run all prompts and evaluate all responses. Defaults to None.
+                If the timeout is hit, this method will throw a `TimeoutError`.
+                Larger values give TLM a higher chance to return outputs for all of your prompts + responses.
+                Smaller values ensure this method does not take too long.
+            retries (int): number of retries to attempt for each individual prompt in case of internal error. Defaults to 1.
+                Larger values give TLM a higher chance of returning outputs for all of your prompts,
+                but this method will also take longer to alert you in cases of an unrecoverable error.
+                Set to 0 to never attempt any retries.
         Returns:
-            float corresponding to the TLM's confidence score
+            float (or list of floats if multiple prompt-responses were provided) corresponding to the TLM's confidence score.
+                    The score quantifies how confident TLM is that the given response is good for the given prompt.
         """
-        return self._event_loop.run_until_complete(
-            self.get_confidence_score_async(
+        if is_collection(prompt):
+            if not is_collection(response):
+                raise ValueError(
+                    "responses must be a list or iterable of strings when prompt is a list or iterable."
+                )
+
+            return self._batch_get_confidence_score(
                 prompt,
                 response,
                 options,
+                timeout=timeout,
+                retries=retries,
             )
-        )
+
+        elif isinstance(prompt, str):
+            if not isinstance(options, dict):
+                raise ValueError("options must be a single TLMOptions object for single prompt.")
```

Comment:
Same comment with not returning the word "TLMOptions" to the user and calling it a dictionary instead